src/Expression/ExpressionBuilder.php

```diff
@@ -17,9 +22,99 @@
  */
 class ExpressionBuilder implements ExpressionBuilderInterface
 {
+    public function __construct(private QueryBuilderInterface|null $queryBuilder = null)
+    {
+    }
+
     public function build(Expression $expression, array &$params = []): string
     {
-        $params = array_merge($params, $expression->getParams());
-        return $expression->__toString();
+        $sql = $expression->__toString();
+        $expressionParams = $expression->getParams();
+
+        if (empty($expressionParams)) {
+            return $sql;
+        }
+
+        if ($this->queryBuilder === null || isset($expressionParams[0])) {
+            $params = array_merge($params, $expressionParams);
```

Comment:
Result params can be incorrect, when `$params` use string keys.

Reply:
For this case the behavior is the same as before.